File: app/src/main/java/com/codenicely/services/feelbazaar/vendorapp/dashboard/provider/RetrofitMyDashboardProvider.java
package com.codenicely.services.feelbazaar.vendorapp.dashboard.provider;

import android.util.Log;

import com.codenicely.services.feelbazaar.vendorapp.dashboard.data.MyDashboardResponse;
import com.codenicely.services.feelbazaar.vendorapp.helper.PresenterCallback;
import com.codenicely.services.feelbazaar.vendorapp.helper.ProviderUtils;
import com.codenicely.services.feelbazaar.vendorapp.helper.WikiEduDashboardApi;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class RetrofitMyDashboardProvider implements MyDashboardProvider {

    private WikiEduDashboardApi wikiEduDashboardApi;

    public RetrofitMyDashboardProvider() {
        wikiEduDashboardApi = ProviderUtils.getApiObject();
    }

    @Override
    public void requestCourseList(String cookies, final PresenterCallback presenterCallback) {
        Call<MyDashboardResponse> courseDetailResponseCall =
                wikiEduDashboardApi.getDashboardDetail("dashboard.json",cookies );
        courseDetailResponseCall.enqueue(new Callback<MyDashboardResponse>() {
            @Override
            public void onResponse(Call<MyDashboardResponse> call, Response<MyDashboardResponse> response) {
                presenterCallback.onSuccess(response.body());
            }

            @Override
            public void onFailure(Call<MyDashboardResponse> call, Throwable t) {
                presenterCallback.onFailure();
            }
        });
    }
}

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/dashboard/RetrofitMyDashboardProvider.kt
package org.wikiedufoundation.wikiedudashboard.ui.dashboard

import org.wikiedufoundation.wikiedudashboard.data.network.WikiEduDashboardApi
import org.wikiedufoundation.wikiedudashboard.ui.dashboard.data.MyDashboardResponse
import org.wikiedufoundation.wikiedudashboard.util.PresenterCallback
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

/**
 * Class that implements [requestCourseList] to get CourseList data
 * ***/
class RetrofitMyDashboardProvider(
    private val wikiEduDashboardApi: WikiEduDashboardApi
) : MyDashboardContract.Provider {

    override fun requestCourseList(cookies: String, presenterCallback: PresenterCallback<MyDashboardResponse>) {
        val courseDetailResponseCall = wikiEduDashboardApi.getDashboardDetail(cookies)
        courseDetailResponseCall.enqueue(object : Callback<MyDashboardResponse> {
            override fun onResponse(call: Call<MyDashboardResponse>, response: Response<MyDashboardResponse>) {

                presenterCallback.onSuccess(response.body())
            }

            override fun onFailure(call: Call<MyDashboardResponse>, t: Throwable) {
                presenterCallback.onFailure()
            }
        })
    }
}
